src/lex/mod.rs

```diff
@@ -374,31 +373,21 @@ impl Lexer {
             self.next_char();
         }
         let float = if hex {
-            hexf_parse::parse_hexf64(&buf, false).map_err(|err| err.to_string())
-        } else {
-            buf.parse()
-                .map_err(|err: std::num::ParseFloatError| err.to_string())
-        }?;
-        if float.is_infinite() {
-            return Err("overflow parsing floating literal".into());
-        }
-        let should_be_zero = buf.bytes().all(|c| match c {
-            b'.' | b'+' | b'-' | b'e' | b'p' | b'0' => true,
-            _ => false,
-        });
-        if float == 0.0 && !should_be_zero {
-            Err("underflow parsing floating literal".into())
```

Comment:
I know we decided to get rid of the check for `INF` - is there a reason you also got rid of the check for underflow?